src/jv_aux.c

```diff
@@ -427,6 +427,70 @@ jv jv_setpath(jv root, jv path, jv value) {
   return jv_set(root, pathcurr, jv_setpath(subroot, pathrest, value));
 }
 
+jv jv_addpath(jv root, jv path, jv add){
+	if(jv_get_kind(path) != JV_KIND_ARRAY || !jv_is_valid(add)){
+		jv_free(root);
+		jv_free(path);
+		jv_free(add);
+		return jv_invalid();
+	}
+
+	if(jv_get_kind(root) != JV_KIND_OBJECT && jv_get_kind(root) != JV_KIND_ARRAY){
+		jv_free(root);
+
+		if(!jv_equal(jv_copy(path), jv_array())){
+			jv_free(path);
+			jv_free(add);
+			return jv_invalid();
+		}
+
+		jv_free(path);
+
+		return add;
```

Comment:
I don't get this case.

Also, what if `root` is a `jv_null()`?  Should we treat it as being a container-like value, like `jq` does?